src/middlewared/middlewared/plugins/backup.py

```diff
@@ -87,17 +90,24 @@ class BackupService(CRUDService):
     async def query(self, filters=None, options=None):
         return await self.middleware.call('datastore.query', 'tasks.cloudsync', filters, options)
 
-    async def _clean_credential(self, data):
+    async def _clean_credential(self, verrors, name, data):
+
         credential = await self.middleware.call('datastore.query', 'system.cloudcredentials', [('id', '=', data['credential'])], {'get': True})
-        assert credential is not None
+        if credential is None:
+            verrors.add(f'{name}.credential', f'Credential {data["credential"]} not found', errno.EEXIST)
```

Comment:
Again, I don't think this is a good idea to report "File already exists" in this case.

Reply:
@themylogin right, changed to ENOENT